### PR TITLE
Record user email in New Relic

### PR DESCRIPTION
**Pull request by dmohs**

### CODE REVIEW

**Comment by zarsky-broad** [comment]
Do we have to? Personally identifiable information makes me nervous. Could we hash it and store that?

File: src/libs/utils.js [hunk 1]
```diff
@@ -88,3 +88,26 @@ export const memoizeWithTimeout = (fn, resolver, ms) => {
...
   }
 }
+
+
+/**
+ * Returns an insecure hash from a string.
+ * Stolen from:
+ * https://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript-jquery
+ */
+export const hashFnv32a = (str, asString, seed) => {
```
Comment (zarsky-broad) [inline src/libs/utils.js]: Use seed = x to set a default value

File: src/libs/utils.js [hunk 2]
```diff
@@ -88,3 +88,26 @@ export const memoizeWithTimeout = (fn, resolver, ms) => {
...
+  /*jshint bitwise:false */
+  var i
+  var l
+  var hval = (seed === undefined) ? 0x811c9dc5 : seed
+
+  for (i = 0, l = str.length; i < l; i++) {
```
Comment (zarsky-broad) [inline src/libs/utils.js]: I think you can just do for i in str.length

**Review by zarsky-broad — APPROVED** [review]

**Comment by dmohs** [comment]
@zarsky-broad PTAL

**Comment by zarsky-broad** [comment]
Generally, I'm ok with trusting developers for this, but when it comes to user privacy, I'd like to provide a guarantee. Consider this: https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest

**Comment by dmohs** [comment]
It doesn't matter how secure the algorithm is if I have access to the IDs since I can just use the same algorithm to hash the IDs and then compare the hashed IDs to the analytics data.

**Comment by dmohs** [comment]
Also, credit goes to Brett for figuring out that hashing doesn't help.

**Comment by zarsky-broad** [comment]
Hmm, I guess it is what it is.

**Comment by dmohs** [comment]
Yeah, it's not ideal, but at least this way we're transparent about it. A hash just makes it look like we're collecting less data even though we're not. That's worse.

**Pull request merged**